### PR TITLE
Update dependency versions

### LINKED ISSUE #125: SageMaker Operator Types fails KubeBuilder Pattern validation check
**What happened**:
SageMaker Operator Types, when included as part of KubeBuilder V2 custom CRD definition fail due to validation errors of unescaped regex patterns. 

```
/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:488:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:110:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:82:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:103:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:466:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:450:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:500:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:515:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:500:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:450:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:82:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:466:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:103:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:488:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:515:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
/go/pkg/mod/github.com/aws/amazon-sagemaker-operator-for-k8s@v1.0.1-0.20200410212604-780c48ecb21a/api/v1/common/sagemaker_api.go:110:2: extra arguments provided: "://([^/]+)/?(.*)$" (at <input>:1:12)
```

**What you expected to happen**:
KubeBuilder should generate CRD specification which includes AWS SageMaker Operator Types

**How to reproduce it (as minimally and precisely as possible)**:
```
import (
	commonv1 "github.com/aws/amazon-sagemaker-operator-for-k8s/api/v1/common"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

// GuestbookSpec defines the desired state of Guestbook
type GuestbookSpec struct {
	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
	// Important: Run "make" to regenerate code after modifying this file

	AlgorithmSpecification *commonv1.AlgorithmSpecification `json:"algorithmSpecification"`

	EnableInterContainerTrafficEncryption *bool `json:"enableInterContainerTrafficEncryption,omitempty"`

	EnableNetworkIsolation *bool `json:"enableNetworkIsolation,omitempty"`
...
//Run make install with above  types in custom operator
make install 
```

**Anything else we need to know?**:
Tried copying the above types and escaped the regex pattern with quotes (``// +kubebuilder:validation:Pattern='^(https|s3)://([^/]+)/?(.*)$'``) and everything worked

**Environment**:
- Kubernetes version (use `kubectl version`):Version: version.Version{KubeBuilderVersion:"2.3.1", KubernetesVendor:"1.16.4", GitCommit:"8b53abeb4280186e494b726edf8f54ca7aa64a49", BuildDate:"2020-03-26T16:42:00Z", GoOs:"unknown", GoArch:"unknown"}
- Operator version (controller image tag):	github.com/aws/amazon-sagemaker-operator-for-k8s v1.0.1-0.20200410212604-780c48ecb21a
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Installation method:
- Others:

### PR DESCRIPTION
**Pull request by RedbackThomson**
### What does this PR do / how does this improve the operators?
Updates the following Go dependencies:
- `controller-gen`: `v0.2.0-beta.2` -> `v0.3.0`
- `aws-sdk-go-v2`: `v0.20.0` -> `v0.24.0`
- `controller-runtime`: `v0.2.0` -> `v0.6.1`
- `client-go`: `v11.0.1-#` -> `v0.18.6`

### Which issue(s) does this PR fix?

Fixes #125 

### Special notes for the reviewer:
Had to make a small change to each of the `suite_test.go` files (in `/api/*` and `/controllers/*`) because the `NewLineReporter` moved to the `envtest/printer` directory. Also had to create a new unit test helper method `CreateMockNamespace` which creates a namespace within the mock API server. It seems that the newer version of the `controller-runtime` require the namespace be created before attempting to use it. All `make` targets work properly.

### Does this PR require changes to documentation?
No

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you written or refactored unit tests to cover the change?
* [x] Have you ran all unit tests and ensured they are passing?
* [x] Have you manually tested each feature that is being added/modified?
* [x] Have you ensured you have not introduced linting errors?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: api/v1/batchtransformjob/batchtransformjob_types.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ type BatchTransformJobSpec struct {
 	Region *string `json:"region"`
 
 	// A custom SageMaker endpoint to use when communicating with SageMaker.
-	// +kubebuilder:validation:Pattern=^(https|http)://.*$
+	// +kubebuilder:validation:Pattern="^(https|http)://.*$"
```
Comment (mbaijal) [inline api/v1/batchtransformjob/batchtransformjob_types.go:59]: Were you able to verify that was actually a version mismatch issue ?

File: config/base/kustomization.yaml [hunk 2]
```diff
@@ -14,3 +12,5 @@ images:
   newTag: v1
 resources:
 - ../manager
+patchesStrategicMerge:
+- manager_auth_proxy_patch.yaml
```
Comment (mbaijal) [inline config/base/kustomization.yaml:16]: why the change

**Comment by mbaijal** [comment]
We should probably call out these versions in the docs or atleast the READMEs too

**Comment by RedbackThomson** [comment]
> We should probably call out these versions in the docs or atleast the READMEs too

These versions are specified in the `go.mod` file? Which versions would you otherwise want to explicitly state?

File: config/base/kustomization.yaml (hunk 2 above)
Comment (RedbackThomson) [reply config/base/kustomization.yaml:16]: This actually comes from @akartsky 's latest changes. I switched the order of his `make` targets so that it runs the US target last, that way all of our Kustomize templates have the US implementation as the default when checked in.

File: api/v1/batchtransformjob/batchtransformjob_types.go (hunk 1 above)
Comment (RedbackThomson) [reply api/v1/batchtransformjob/batchtransformjob_types.go:59]: Yes. This error was exactly as predicted.

File: config/base/kustomization.yaml (hunk 2 above)
Comment (mbaijal) [reply config/base/kustomization.yaml:16]: wow, good thinking.

**Comment by mbaijal** [comment]
> > We should probably call out these versions in the docs or atleast the READMEs too
> 
> These versions are specified in the `go.mod` file? Which versions would you otherwise want to explicitly state?

I guess that's true. If someone is developing on top of these operators they would be familiar enough with golang to know where to look.
